src/mlpack/core/tree/cover_tree/traits.hpp

```diff
@@ -33,6 +33,12 @@ class TreeTraits<CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>>
   static const bool HasOverlappingChildren = true;
 
   /**
+   * Cover trees do have self-children, so points can be included in more than
+   * one node.
+   */
+  static const bool HasDuplicatedPoints = true;
```

Comment:
I think that `HasDuplicatedPoints` was meant to encode whether or not `NumDescendants()` accounted for each point uniquely.  In the case of the cover tree, even though points are duplicated, the properties of the self-child (and the fact that it only holds one point) mean that we can correctly calculate the number of unique descendants and return them with `Descendant()`.  So maybe our name `HasDuplicatedPoints` should be changed somehow, but either way, I think that this value should be `false`.